edx_rest_framework_extensions/permissions.py

```diff
@@ -18,15 +16,44 @@ def has_permission(self, request, view):
         return request.user and request.user.is_superuser
 
 
+class IsStaff(BasePermission):
+    """
+    Allows access to "global" staff users..
+    """
+    def has_permission(self, request, view):
+        return request.user.is_staff
+
+
+class IsUserInUrl(BasePermission):
+    """
+    Allows access if the the requesting user matches the user in the URL.
+    """
+    def has_permission(self, request, view):
+        user_parameter_name = 'username'
+        url_username = (
+            request.GET.get(user_parameter_name, '') or
+            getattr(request, 'parser_context', {}).get('kwargs', {}).get(user_parameter_name, '')
```

Comment:
This might be silly, but if a url had both 'username' and this custom `user_parameter_name`, shouldn't this one take precedence?

Reply:
Not sure I follow. What do you mean by "custom" `user_parameter_name`? I think using the variable is just a convenience so that the literal string does not have to be typed out twice in this function right?

Reply:
Um. Yes. I don't know how my brain refactored that code.  Ignore this comment. :)